src/controllers/dynakube/activegate/internal/statefulset/builder/modifiers/kubemon.go

```diff
@@ -72,11 +73,11 @@ func (mod KubernetesMonitoringModifier) getInitContainers() []corev1.Container {
 			Args:            []string{"-c", k8scrt2jksPath},
 			VolumeMounts:    volumeMounts,
 			Resources:       mod.capability.Properties().Resources,
-			SecurityContext: &corev1.SecurityContext{
-				ReadOnlyRootFilesystem: &readOnlyRootFs,
-			},
+			SecurityContext: consts.ContainerSecurityContext.DeepCopy(),
```

Comment:
Did you test if it is fine using the same security context on the container as on the pod?

Reply:
Only some of ContainerSecurityContext settings can be also set on the pod level
```
	PodSecurityContext = corev1.PodSecurityContext{
		// Privileged:               address.Of(false),
		// AllowPrivilegeEscalation: address.Of(false),
		RunAsNonRoot: address.Of(true),
		RunAsUser:    address.Of(DockerImageUser),
		RunAsGroup:   address.Of(DockerImageGroup),
		// Capabilities: &corev1.Capabilities{
		//	Drop: []corev1.Capability{
		//		"ALL",
		//	},
		// },
		SeccompProfile: &corev1.SeccompProfile{
			Type: corev1.SeccompProfileTypeRuntimeDefault,
		},
		//ReadOnlyRootFilesystem: address.Of(false),
	}
```
so I decided to keep them all in the shared ContainerSecurityContext structure.

I have tested a case with `PodSecurityContext` defined and istio enabled. Everything is fine (istio containers define own SecurityContexts with `runAsUser: 0` or `runAsUser: 1337`).